scripts/migrate_assets.sh

```diff
@@ -51,17 +51,67 @@ while getopts ":c:s:t:" opt; do
   esac
 done
 OPTIND=$__optind__
+[ -n "$target" ] && install=true || install=false
+
+if $install; then
+    CUDA_QUANTUM_PATH="$target" && mkdir -p "$CUDA_QUANTUM_PATH"
+    CUDAQ_INSTALL_PREFIX=`dirname "$(readlink -f "${BASH_SOURCE[0]}")"`
+elif [ -z "$CUDA_QUANTUM_PATH" ] && [ -z "$CUDAQ_INSTALL_PREFIX" ]; then 
+    echo -e "\e[01;31mError: Neither CUDAQ_INSTALL_PREFIX nor CUDA_QUANTUM_PATH are defined.\e[0m" >&2
+    (return 0 2>/dev/null) && return 1 || exit 1
+else
+    CUDA_QUANTUM_PATH=${CUDA_QUANTUM_PATH:-"$CUDAQ_INSTALL_PREFIX"}
+    CUDAQ_INSTALL_PREFIX=${CUDAQ_INSTALL_PREFIX:-"$CUDA_QUANTUM_PATH"}
+fi
+
+assets="${source:-$CUDAQ_INSTALL_PREFIX}"
+build_config="${config:-$assets/build_config.xml}"
+if [ ! -f "$build_config" ]; then 
+    build_config="$CUDAQ_INSTALL_PREFIX/build_config.xml"
+fi
+if [ ! -f "$build_config" ]; then
+    echo -e "\e[01;31mError: Missing build configuration.\e[0m" >&2
+    (return 0 2>/dev/null) && return 1 || exit 1
+fi
+
+remove_assets="$CUDA_QUANTUM_PATH/$($install && echo uninstall.sh || echo remove_assets.sh)"
+echo "Migrating assets in $assets."
+echo "Using build configuration $build_config."
+echo "The script to remove the migrated files can be found in $remove_assets."
+
+read -r -d '' confirmation_prompt << 'EOP'
+if [ "$1" == "-y" ]; then continue=true
+else
+    while true; do
+        read -p "Continue (y/n)?" -r choice < /dev/tty
+        case "$choice" in
+            y|Y ) continue=true && break;;
+            n|N ) continue=false && break;;
+            * ) echo "Please enter y or n.";;
+        esac
+    done
+fi
+EOP
 
 function move_artifacts {
     mkdir -p "$2" -m 755 && cd "$1"
     echo "Updating $2 with artifacts in $1:"
+
+    # Prompting for confirmation to remove the copied files in that folder:
+    echo 'echo "Cleaning up '$2':"' >> "$remove_assets"
+    echo "$confirmation_prompt" >> "$remove_assets"
+    echo 'if $continue; then' >> "$remove_assets"
+
     for file in `find . -type f`; 
```

Comment:
I know you didn't change this for this review, but I happened to see it and think about it (regarding filenames with spaces). I believe the following change would handle filenames with spaces better (but feel free to save for later if desired).
```suggestion
    find . -type f -print0 | while IFS= read -r -d '' file; do
```